### DISCUSSION TITLE
Specific Domain Design

### DISCUSSION THREAD

**Discussion by ShaikhaTheGreen** [lululxvi/deepxde#740] [General]
Thanks @lululxvi for the amazing library!
I used it to model a 1D wave equation:
```
def pde(x, u):
    c = 1
    du_xx = dde.grad.hessian(u, x, i=0, j=0)
    du_tt = dde.grad.hessian(u, x, i=1, j=1)
    return   du_tt - (c ** 2)*(du_xx) 
```

I achieved very good results in this geomtime domain:
```
geom = dde.geometry.Interval(0,1)
timedomain = dde.geometry.TimeDomain(0,1)
geomtime = dde.geometry.GeometryXTime(geom, timedomain)
```
The results legend uniform1, uniform2, ... is only indicating the points distribution and number of experiment, so you can ignore them.

![0](https://user-images.githubusercontent.com/25299510/174110135-9bce5a2f-1a11-43e0-a8d4-7c120b7c88eb.png)
![2](https://user-images.githubusercontent.com/25299510/174110192-5a32f715-409d-434b-a184-08a622b5d54d.png)
![4](https://user-images.githubusercontent.com/25299510/174110224-e12deed9-c26b-46f2-a46d-0c4fc75fd648.png)
![6](https://user-images.githubusercontent.com/25299510/174110252-1315ac9f-5137-437d-bfa6-af20215f8162.png)
![8](https://user-images.githubusercontent.com/25299510/174110304-e752111e-baca-4ed8-b982-4c4d99a6fd64.png)
![10](https://user-images.githubusercontent.com/25299510/174110328-122453e0-a549-445c-b4b9-05f719e8740c.png)

My question now is: how can I change the domain to model the wave travelling in a human tissue depth of 10cm and 10mm of bone for example? How can I have multiple units of space involved and have accurate results?

How can I move from my current setup to this multi-region domain?

**Comment by lululxvi** [comment]
Have no idea of what a human tissue depth of 10cm and 10mm of bone is. Your question seems a research problem.

**Reply by ShaikhaTheGreen** [reply]
To simplify, you can imagine this example sketch in a 2D domain:

I would like to utilize PINN to model the wave equation with a source at the point indicated. My questions are:
![Sketch 2022-06-19 16_45_31](https://user-images.githubusercontent.com/25299510/174491728-bb7caee5-b196-476a-86f5-cabf9ce20c1f.png)

1. How can I specify that the distance in the first region is 10 cm and 2cm in the second region?
2. In each region the wave speed (c) is different, how can I implement this in `pde(x,u)`?
3. If it is simpler to implement in 1D and then extend to 2D please advise on how to tackle this setup problem?


Thank you so much for your continuous support I really appreciate your efforts!

**Reply by ragusa** [reply]
I think the question could be about how to define heterogenous zones inside a domain. I would create a domain of size 12 and would use a mask
c = ctissue * (x<10) + cbone *(x>=10)
where ctissue and cbone are the material values you want.

**Reply by ShaikhaTheGreen** [reply]
Thanks for the reply!
What about the units in centimeters or millimeters? How can you control it in the domain?

**Reply by ragusa** [reply]
as with all mathematical modeling, the units have to be consistent so that each term of your PDE has the same units: ddu/dtt should have the same units as c*ddu/ddx.
